app-example/src/main/java/com/badoo/ribs/example/app_bar/AppBarInteractor.kt

```diff
@@ -0,0 +1,35 @@
+package com.badoo.ribs.example.app_bar
+
+import androidx.lifecycle.Lifecycle
+import com.badoo.mvicore.android.lifecycle.startStop
+import com.badoo.mvicore.binder.using
+import com.badoo.ribs.clienthelper.interactor.Interactor
+import com.badoo.ribs.core.modality.BuildParams
+import com.badoo.ribs.example.app_bar.AppBarBuilder.Params
+import com.badoo.ribs.example.app_bar.mapper.UserToViewModel
+import com.badoo.ribs.example.app_bar.mapper.ViewEventToOutput
+import com.badoo.ribs.example.network.model.User
+import com.badoo.ribs.example.repository.UserRepository
+import io.reactivex.ObservableSource
+import io.reactivex.functions.Consumer
+
+internal class AppBarInteractor(
+    buildParams: BuildParams<Params>,
+    userRepository: UserRepository,
+    private val output: Consumer<AppBar.Output>
+) : Interactor<AppBar, AppBarView>(
+    buildParams = buildParams
+) {
+
+    private val userSource: ObservableSource<User> =
```

Comment:
What will happen if UserRepository will send onError? How will it handle when user logges in/out

Reply:
As far as I understand, this RIB will only be used in logged in scope of the app, so we will never have to deal with a logged out user.

As for the error, do we expect UserRepository to ever send onError? If user is not found, UserRepository should just send Maybe.empty() and we will continue showing the placeholder in the AppBar.

Reply:
> As far as I understand, this RIB will only be used in logged in scope of the app

While this is right, I wouldn't encourage building on similar "invisible" assumptions in general. Requirements should be expressed as dependencies, other than that a module should be reusable in any context.

> As for the error, do we expect UserRepository to ever send onError? 

+1? @antonshilov

Reply:
@zsoltk 
> While this is right, I wouldn't encourage building on similar "invisible" assumptions in general.

The method in UserRepository to fetch a user is a Maybe<User> method, so no hard assumption is made that a user object will always be present. In case the user is logged out, the way the RIB should work is that the UserRepository should return Maybe.empty() and then the AppBar will continue to use placeholder instead of user avatar.

Reply:
I meant the assumptions like "it will be only used in logged in scope of the app" are what shouldn't be made, and even that just a general piece of advice.

I'm ok with the actual impl in general.

Reply:
By the logged out user I meant Anonymous user, maybe we could explicitly handle the case for that. But returning no user from the repo will work too

> As for the error, do we expect UserRepository to ever send onError? If user is not found, UserRepository should just send Maybe.empty() and we will continue showing the placeholder in the AppBar.

I am expecting UserRepo implementation to get user from the network and this could cause errors, so we will need to handle them in the repo or somwhere else.

Reply:
Perhaps for now it would be best to just return empty user from the repository and as and when we implement the repository we can decide how we want to handle the anonymous user?

Reply:
Yep, agree lets leave it to the implementation.